server/route.go

```diff
@@ -584,24 +587,32 @@ func (s *Server) sendLocalSubsToRoute(route *client) {
 	var raw [4096]*subscription
 	subs := raw[:0]
 
-	s.sl.localSubs(&subs)
+	closed := false
+	threshold := int64(1024 * 1024)
 
 	route.mu.Lock()
+	s.sl.localSubs(&subs)
+	route.Noticef("Sending %v local subscriptions to route", len(subs))
 	for _, sub := range subs {
 		// Send SUB interest only if subject has a match in import permissions
 		if !route.canImport(sub.subject) {
 			continue
 		}
 		proto := fmt.Sprintf(subProto, sub.subject, sub.queue, routeSid(sub))
 		route.queueOutbound([]byte(proto))
-		if route.out.pb > int64(route.out.sz*2) {
-			route.flushSignal()
+		if route.out.pb > threshold {
+			route.flushOutbound()
```

Comment:
Should check to see if this actually did anything and if not do a signal.

Reply:
not needed because it returns false only in one condition: the flushOutbound flag is already set and in that case the lock is not released, and then we signal at the end of this loop.

Reply:
It could be released, but if you signal at the end of the loop and you don't think you need a kick in between then its ok by me.

Reply:
Actually I think we should do 2 things: signal and release/reacquire lock. If we don't, here is what is happening. Say we are in this loop and call flush because we have accumulated some data. Another go routine is in flush (and the flag is set) and has released the lock to do the actual socket write. If we don't signal/release lock/reacquire lock, then this loop will continue to enqueue and every time it calls flushOutbound() it would get false: the other go-routine will not be able to grab the lock after the socket write.
The point is that simply signaling is not going to help if we don't release the lock.

Reply:
Agree.

Reply:
will calling runtime.Gosched() in between be ok or do you prefer a very small sleep (if so, what value do you think is appropriate)?

Reply:
If you release the lock that will engage the scheduler. No need IMO for explicit call or a sleep.